### PR TITLE
flagging-posts and filtering for latest and popular posts

### PR DESCRIPTION
**Pull request by akshyta**

### CODE REVIEW

File: projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.ts [hunk 1]
```diff
@@ -42,23 +48,20 @@ export class FeedItemDetailsComponent implements OnInit {
...
+
+  // sendFlag(feedItemId) {
+  //   this.discussionChatChannel.sendData(
+  //     this.discussionChatChannel.ACTIONS.FLAG,
+  //     {
+  //       feed_item_id: feedItemId
+  //     }
+  //   );
+  //   console.log("sent");
+  // }
```
Comment (anushika99) [inline projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.ts]: Remove comments

File: projects/shared-components/services/s-flags.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return this.http.get<any>(
+      this.apiRoutesService.getRoute(API_ROUTES.FLAGS.PUBLIC.COUNT), {params});
+  }
+
+//   pGetVoters(votableType, votableId, page, count): Observable<IUsers> {
+//     const params = new HttpParams().set('votable_type', votableType).set('votable_id', votableId).set('page', page).set('count', count);
+//     return this.http.get<IUsers>(
+//       this.apiRoutesService.getRoute(API_ROUTES.VOTES.PUBLIC.VOTERS), {params});
+//   }
+
```
Comment (anushika99) [inline projects/shared-components/services/s-flags.service.ts]: Remove comments

File: projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.html [hunk 3]
```diff
@@ -42,17 +42,24 @@ <h3 class="name">
...
           <p
-            [ngClass]="{ 'error-text': -1 > 0 }"
+            [ngClass]="{ 'error-text': 2 > 0 }"
             nbTooltip="Report this message. (3 reports will delete it)"
             nbTooltipStatus="warning"
           >
             <span *ngIf="message.flags_count > 0">{{ feedItem.flags_count }}</span>
             <span> 1 </span>
             <nb-icon icon="flag"></nb-icon>
           </p>
```
Comment (anushika99) [inline projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.html]: Remove

File: projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.ts (hunk 1 above)
Comment (akshyta) [reply projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.ts]: Done

File: projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.html (hunk 3 above)
Comment (akshyta) [reply projects/commudle-admin/src/app/feature-modules/external-feed/components/feed-item-details/feed-item-details.component.html]: Done

File: projects/shared-components/services/s-flags.service.ts (hunk 2 above)
Comment (akshyta) [reply projects/shared-components/services/s-flags.service.ts]: Done

**Review by anushika99 — APPROVED** [review]

**Pull request merged**